src/index.test.ts

```diff
@@ -704,6 +704,64 @@ https://example.com
         expect(value).toContain("https://custom.com/image.png");
       });
 
+      test("should pass URL parameter to ogTransformer", async () => {
+        const markdown = `
+https://example.com/path?param=value
+`;
+
+        const ogTransformerSpy = vi.fn((og, url) => ({
+          ...og,
+          title: `Title for ${url.hostname}`,
+          description: `Path: ${url.pathname}`,
+        }));
+
+        const { value } = await remark()
+          .use(remarkLinkCard, {
+            ogTransformer: ogTransformerSpy,
+          })
+          .process(markdown);
+
+        expect(ogTransformerSpy).toHaveBeenCalledWith(
+          expect.objectContaining({
+            title: "Test Site Title",
+            description: "Test Description",
+          }),
+          new URL("https://example.com/path?param=value"),
+        );
+
+        expect(value).toContain("Title for example.com");
+        expect(value).toContain("Path: /path");
+      });
+
+      test("should use URL information in ogTransformer for conditional transformation", async () => {
+        const markdown = `
+https://github.com/user/repo
+
+https://example.com/page
+`;
+
+        const { value } = await remark()
+          .use(remarkLinkCard, {
+            ogTransformer: (og, url) => {
+              if (url.hostname === "github.com") {
+                return {
+                  ...og,
+                  title: `GitHub: ${og.title}`,
+                  description: "This is a GitHub repository",
+                };
+              }
+              return og;
+            },
+          })
+          .process(markdown);
+
+        expect(value).toContain("GitHub: Test Site Title");
+        expect(value).toContain("This is a GitHub repository");
+        // example.com should not be modified
+        expect(value).toContain("Test Site Title");
```

Comment:
This assertion also matches the GitHub title, so it could be made a bit more strict to avoid false positives.